From 67f40c35be9c915ac75bddfaf837ffbaf4b42efb Mon Sep 17 00:00:00 2001
From: Shawn Ferry <sferry@gmail.com>
Date: Thu, 11 May 2017 15:26:20 -0400
Subject: [PATCH] Solaris Patches for client.cfg and server.cfg

Changes are not suitable for upstream

Changes needed, mainly :
    Making use of RabbitMQ (default is ActiveMQ)
    The default port used to connect to RabbitMQ is 61613
    Default user : 'mcollective' & passwd : 'changeme'
    Change RabbitMQ to log to /dev/stderr, so it goes to the SMF log
    Set the PSK type explicitly to UID and the default to 'changeme'
---
 etc/client.cfg.dist | 23 +++++++++++++----------
 etc/server.cfg.dist | 23 +++++++++++++----------
 2 files changed, 26 insertions(+), 20 deletions(-)

diff --git a/etc/client.cfg.dist b/etc/client.cfg.dist
index 1acffee..dd6960b 100644
--- a/etc/client.cfg.dist
+++ b/etc/client.cfg.dist
@@ -1,21 +1,24 @@
 main_collective = mcollective
 collectives = mcollective
-libdir = /usr/libexec/mcollective
-logger_type = console
 loglevel = warn
+libdir = /usr/share/mcollective
+logger_type = file
+# Send output to SMF service log
+logfile = /dev/stderr
 
 # Plugins
 securityprovider = psk
-plugin.psk = unset
+plugin.psk = changeme
+plugin.psk.callertype = uid
 
-connector = activemq
-plugin.activemq.pool.size = 1
-plugin.activemq.pool.1.host = stomp1
-plugin.activemq.pool.1.port = 6163
-plugin.activemq.pool.1.user = mcollective
-plugin.activemq.pool.1.password = marionette
+connector = rabbitmq
+plugin.rabbitmq.vhost = /mco
+plugin.rabbitmq.pool.size = 1
+plugin.rabbitmq.pool.1.host = localhost
+plugin.rabbitmq.pool.1.port = 61613
+plugin.rabbitmq.pool.1.user = mcollective
+plugin.rabbitmq.pool.1.password = changeme
 
 # Facts
 factsource = yaml
 plugin.yaml = /etc/mcollective/facts.yaml
-
diff --git a/etc/server.cfg.dist b/etc/server.cfg.dist
index 2038324..2ced9c2 100644
--- a/etc/server.cfg.dist
+++ b/etc/server.cfg.dist
@@ -1,22 +1,25 @@
 main_collective = mcollective
 collectives = mcollective
-libdir = /usr/libexec/mcollective
-logfile = /var/log/mcollective.log
+libdir = /usr/share/mcollective
+logger_type = file
+# Send output to SMF service log
+logfile = /dev/stderr
 loglevel = info
 daemonize = 1
 
 # Plugins
 securityprovider = psk
-plugin.psk = unset
+plugin.psk = changeme
+plugin.psk.callertype = uid
 
-connector = activemq
-plugin.activemq.pool.size = 1
-plugin.activemq.pool.1.host = stomp1
-plugin.activemq.pool.1.port = 6163
-plugin.activemq.pool.1.user = mcollective
-plugin.activemq.pool.1.password = marionette
+connector = rabbitmq
+plugin.rabbitmq.vhost = /mco
+plugin.rabbitmq.pool.size = 1
+plugin.rabbitmq.pool.1.host = localhost
+plugin.rabbitmq.pool.1.port = 61613
+plugin.rabbitmq.pool.1.user = mcollective
+plugin.rabbitmq.pool.1.password = changeme
 
 # Facts
 factsource = yaml
 plugin.yaml = /etc/mcollective/facts.yaml
-
-- 
2.11.0 (Apple Git-81)

